bitcoin/src/blockdata/transaction.rs

```diff
@@ -833,6 +834,124 @@ impl Transaction {
     pub fn script_pubkey_lens(&self) -> impl Iterator<Item = usize> + '_ {
         self.output.iter().map(|txout| txout.script_pubkey.len())
     }
+
+    // TODO: Find a better reference for how taproot sigops are limited.
```

Comment:
In b5347f8d754d6c2daa287d838af5dcfde1910cad:

The provided reference (which is by the primary Taproot author, and IIRC is the person who came up with the "each sigop costs 50 weight" idea) seems fine to me.

Reply:
I think this function could be used for Taproot BTW. Its interpretation would be a bit different but that's fine.

We would need to update `count_sigops_internal` to count CHECKSIGADD. (Which it currently looks like this PR never counts? Though the doccomments imply that it will, by saynig that the legacy functions *don't*.)

Reply:
> I think this function could be used for Taproot BTW

This function is a Transaction level function, but Taproot scripts don't count toward the transaction's sigop count.

We could add a taproot variant to the `Script` level count function, though. Since the only time sigops are tracked during taproot verification is when running the tapscript, the script interpreter holds a running budget that it ticks down every time it performs SIGADD etc.

We could also then build upon that and add a `Witness` method that would calculate the budget (which is serialized witness stack size + 50) and see if the sigops of the tapscript go over that.

Reply:
Either way, I think this is a separate PR, so I will resolve this for now.